rackcity/views/power_views.py

```diff
@@ -596,30 +558,75 @@ def make_bcman_request(chassis, blade, power_command):
     return result, exit_status
 
 
-def get_chassis_power_request_parameters(data):
+def get_chassis_power_request_parameters(request):
+    data = JSONParser().parse(request)
     if ("chassis_id" not in data) or ("blade_slot" not in data):
-        raise ChassisPowerManagementException(
+        raise PowerManagementException(
             "Must specify 'chassis_id' and 'blade_slot' on chassis power request."
         )
     try:
         blade_slot = int(data["blade_slot"])
         chassis_id = int(data["chassis_id"])
     except ValueError:
-        raise ChassisPowerManagementException(
+        raise PowerManagementException(
             "Parameters 'chassis_id' and 'blade_slot' must be of type int."
         )
     try:
         chassis = Asset.objects.get(id=chassis_id)
     except ObjectDoesNotExist:
-        raise ChassisPowerManagementException(
-            "Chassis" + GenericFailure.DOES_NOT_EXIST.value
+        raise PowerManagementException("Chassis" + GenericFailure.DOES_NOT_EXIST.value)
+    if not user_has_power_permission(request.user, asset=chassis):
+        raise UserPowerPermissionException(
+            "User "
+            + request.user.username
+            + " does not have power permission and does not own asset with id="
+            + str(data["id"])
         )
-    if (not chassis.model.is_blade_chassis) or (chassis.model.vendor != "BMI"):
-        raise ChassisPowerManagementException(
-            "Power is only network controllable for blade chassis of vendor 'BMI'."
+    if not is_asset_power_controllable_by_bcman(chassis):
+        raise PowerManagementException(
+            "Power is only network controllable for blade chassis of vendor 'BMI' with hostname not in storage."
         )
     if (blade_slot < 1) or (blade_slot > 14):
-        raise ChassisPowerManagementException(
+        raise PowerManagementException(
             "Blade slot " + str(blade_slot) + " does not exist on chassis."
         )
-    return chassis.hostname, blade_slot
+    return chassis, blade_slot
+
+
+def get_pdu_power_request_parameters(request):
+    data = JSONParser().parse(request)
+    if "id" not in data:
+        raise PowerManagementException("Must specify 'id' on pdu power request.")
+    try:
+        asset = Asset.objects.get(id=data["id"])
+    except ObjectDoesNotExist:
+        raise PowerManagementException("Asset" + GenericFailure.DOES_NOT_EXIST.value)
```

Comment:
This I think should have a space

Reply:
After Asset

Reply:
ACTUALLY i did something crazy and made that enum equal to `DOES_NOT_EXIST = " does not exist."` so you never have to add a space

Reply:
CRAZY